File: app/src/main/java/com/mikepenz/materialdrawer/app/DrawerItems/CustomPrimaryDrawerItem.java
package com.mikepenz.materialdrawer.app.drawerItems;

import android.support.v7.widget.RecyclerView;

import com.mikepenz.materialdrawer.holder.ColorHolder;
import com.mikepenz.materialdrawer.model.PrimaryDrawerItem;

public class CustomPrimaryDrawerItem extends PrimaryDrawerItem {

    private ColorHolder background;

    public CustomPrimaryDrawerItem withBackgroundColor(int backgroundColor) {
        this.background = ColorHolder.fromColor(backgroundColor);
        return this;
    }

    public CustomPrimaryDrawerItem withBackgroundRes(int backgroundRes) {
        this.background = ColorHolder.fromColorRes(backgroundRes);
        return this;
    }

    @Override
    public void bindView(RecyclerView.ViewHolder holder) {
        super.bindView(holder);

        if (background != null) {
            background.applyToBackground(holder.itemView);
        }
    }
}

File: app/src/main/java/com/mikepenz/materialdrawer/app/drawerItems/CustomPrimaryDrawerItem.kt
package com.mikepenz.materialdrawer.app.drawerItems

import com.mikepenz.materialdrawer.holder.ColorHolder
import com.mikepenz.materialdrawer.model.AbstractBadgeableDrawerItem

class CustomPrimaryDrawerItem : AbstractBadgeableDrawerItem<CustomPrimaryDrawerItem>() {

    var background: ColorHolder? = null

    fun withBackgroundColor(backgroundColor: Int): CustomPrimaryDrawerItem {
        this.background = ColorHolder.fromColor(backgroundColor)
        return this
    }

    fun withBackgroundRes(backgroundRes: Int): CustomPrimaryDrawerItem {
        this.background = ColorHolder.fromColorRes(backgroundRes)
        return this
    }

    override fun bindView(holder: AbstractBadgeableDrawerItem.ViewHolder, payloads: List<Any>) {
        super.bindView(holder, payloads)
        background?.applyToBackground(holder.itemView)
    }
}
